### PR TITLE
sam init command help

### PR DESCRIPTION
**Pull request by sriram-mv**
#### Which issue(s) does this change fix?
N/A

#### Why is this change necessary?
* Clean up the help text for sam init, especially its the first command we ask to run after looking at the `sam --help` text.
* Split up examples section to showcase what each section is about, we should do this for other commands as well.

#### How does it address the issue?
init

#### What side effects does this change have?
N/A

* Ran the unit and integration tests for the init suite.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/init/command.py [hunk 1]
```diff
@@ -238,12 +218,12 @@ def wrapped(*args, **kwargs):
...
     "--tracing/--no-tracing",
     default=None,
-    help="Enable AWS X-Ray tracing for your lambda functions",
+    help="Enable AWS X-Ray tracing for application.",
 )
 @click.option(
     "--application-insights/--no-application-insights",
     default=None,
-    help="Enable CloudWatch Application Insights monitoring for your application",
+    help="Enable CloudWatch Application Insights monitoring for application.",
```
Comment (mndeveci) [inline samcli/commands/init/command.py:226]: nit: I remember seeing another PR which removed the trailing dots in a help text, we might need to align which approach we are taking

**Review by mndeveci — APPROVED** [review]

File: samcli/commands/init/command.py (hunk 1 above)
Comment (sriram-mv) [reply samcli/commands/init/command.py:226]: Oh interesting, do we remember the rationale for that? Agree on adopting one vs other in totality for consistency.

Comment (mndeveci) [reply samcli/commands/init/command.py:226]: I think that is more of a personal preference, not sure if CLI recommendations favor one or the other. I am OK with either but just wanted to make sure we will have a persistent help text through the CLI. :)

File: samcli/commands/init/core/command.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            with formatter.indented_section(name=option_heading, extra_indents=1):
+                formatter.write_rd(options.get("extras", [RowDefinition()]))
+                formatter.write_rd(
+                    [RowDefinition(name="", text="\n")]
+                    + [
+                        opt
+                        for options in zip(opts, [RowDefinition(name="", text="\n")] * (len(opts)))
+                        for opt in options
+                    ],
+                )
```
Comment (mildaniel) [inline samcli/commands/init/core/command.py:116]: Is there a reason for not using `CoreCommand._format_options()` here?
https://github.com/aws/aws-sam-cli/blob/develop/samcli/cli/core/command.py#L38

**Review by mildaniel — COMMENTED** [review]
Thanks Sriram, I really like the sectioned examples. Left one small questions.

**Review by mildaniel — APPROVED** [review]

**Pull request merged**